spmd/compiler/api.py

```diff
@@ -298,9 +317,24 @@ def _convert_to_distributed(
             assert i < len(
                 inps
             ), f"got more placeholer nodes ({i + 1}) than inputs ({len(inps)})"
-            node_to_obj[node] = DTensor.from_local(
-                inps[i], schemas[i].mesh, schemas[i].placements
-            )
+            if training_phase == TrainingPhase.FORWARD:
+                # in the forward phase we start with the full "global" ensors.
```

Comment:
nit: enstors -> tensors

Reply:
The changes in _convert_to_distributed are basically meant to fix a bug where we were not picking up changes in non-tensor args.

One of the things that propagate_input_sharding implementations are allowed to do is to modify non-tensor args. Specifically one of the things that we do specially for view ops is to modify the shape input. 

If you look at the implementation for the eager mode propragatrion (here: https://github.com/pytorch/tau/blob/5e4f98b49dcf26254300e66917369049cda84198/spmd/tensor/dispatch.py#L129-L141)

you'll notice that for non-tensor args, it actually uses the args that were modified by the sharding propgataion function implementation.


So what this does in this particular failing example , is that for torch.expand, it changes the input shape from global shape to local shape in the transformed graph.